Assets/MixedRealityToolkit.Extensions/Sharing/SpatialAlignment/Common/ISpatialCoordinate.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Numerics;
+
+namespace Microsoft.MixedReality.Experimental.SpatialAlignment.Common
+{
+    /// <summary>
+    /// Defines the potential states of alignment for a strategy.
+    /// </summary>
+    public enum LocatedState
+    {
+        /// <summary>
+        /// An error has occurred with the alignment strategy.
+        /// </summary>
+        Error = -1,
+
+        /// <summary>
+        /// Alignment has never been resolved in this session.
+        /// </summary>
+        Unresolved = 0,
+
+        //TODO this vs Resolved, what are the use cases.
+        /// <summary>
+        /// Alignment is temporarily inhibited (usually due to some form of tracking loss).
+        /// </summary>
+        Inhibited,
+
+        /// <summary>
+        /// Alignment has been resolved at least once but there is no active tracking.
+        /// </summary>
+        Resolved,
+
+        /// <summary>
+        /// Alignment has been resolved at least once and there is active tracking.
+        /// </summary>
+        Tracking,
+    }
+
+    /// <summary>
+    /// This represents a spatial coordinate that can then be used to convert position and rotation to and from this coordinate space.
+    /// </summary>
+    public interface ISpatialCoordinate
+    {
+        /// <summary>
+        /// Occurs when the value of the <see cref="State"/> property has changed.
+        /// </summary>
+        event Action StateChanged;
+
+        /// <summary>
+        /// Gets the Id representing this coordinate.
+        /// </summary>
+        string Id { get; }
+
+        // TODO agree with Jared
+        //Vector3 Accuracy { get; }
```

Comment:
Seems like finding an accuracy metric might be tough, but it could be very useful

Reply:
In my [Spatial Alignment](http://aka.ms/SpatialAlignment) reference architecture, I [define](https://github.com/jbienzms/SpatialAlignment/blob/Dev/SpatialAlignment-Unity/Assets/SpatialAlignment/Strategies/IAlignmentStrategy.cs) accuracy as a metric of drift in meters. Think of it as "I am confident that this point in Unity space matches the same point in real-world space, plus or minus (x, y, z) meters".

For something like QR, this value could just be hard-coded. It's basically knowing the best possible spatial resolution using the devices camera. (RGB camera frustum projected into real-world space). For something like GPS, this value can come directly from devices which support it. Many bluetooth GPS units report HDOP and VDOP (horizontal and vertical dilution of precision) NMEA sentences. Garmin GPS units have a PGRE NMEA sentance. These can all be used to provide approximate error in meters on all axis in real-time.

Reply:
Do you think it would make more sense to set an enum to define orders of magnitude of precision?

1CM, 10CM, 1M, 10M, etc? I think this would be easier to work with than a Vector3 I'll have to process math on

Reply:
Honestly I'm not a fan of the enum-only approach. Especially if it's only represented as a single unified axis. And if you present the enum on all 3 axis (which I think is important) you're really just trading one type of number for another. One which has less accuracy.

If you feel strongly about the enum though you could easily write an extension method that converts Vector3 to your enum type. It's always possible to go from a higher fidelity type to a lower fidelity one, just not the other way around.

Reply:
You can use enum flags to register the different axies, and store a composite enum that way.

I feel strongly about it not just for ease of use, but for performance and storage reasons.

A vector3 is 32bit * 3, while an enum is a byte or at worse 32bit.
You get everything you need in 99% of usecase, for far less data. All this needs to be networked as well, so keeping data down is important.

Beyond that just parsing the tolerance becomes a lot cheaper than doing vector size checks, which require square root operations and consume a lot of operations.